python/pyspark/context.py

```diff
@@ -956,6 +956,15 @@ def setCheckpointDir(self, dirName):
         """
         self._jsc.sc().setCheckpointDir(dirName)
 
+    def getCheckpointDir(self):
```

Comment:
Let's also decorate this with `@since(3.1)`.

Reply:
Should also add the corresponding type hint below https://github.com/apache/spark/blob/master/python/pyspark/context.pyi#L154 line. cc @zero323 FYI.

Reply:
Just for the reference ‒ it will be just

```python
    def getCheckpointDir(self) -> Optional[str]: ...
```

Reply:
Thanks for the review. Do you know how to properly import ``since`` into ``python/pyspark/context.py``? ``since`` is defined in the ``python/pyspark/__init__.py`` and I haven't been able to import it into ``python/pyspark/context.py`` without getting errors.

Reply:
That's because of circular dependency, right?

I think the only way to make it work is to move 

https://github.com/apache/spark/blob/82721ce00b6cf535abd3d9cd66445e452554d15d/python/pyspark/__init__.py#L53

somewhere after `since` definition, for example here:

https://github.com/apache/spark/blob/82721ce00b6cf535abd3d9cd66445e452554d15d/python/pyspark/__init__.py#L116-L117

I am not sure if we want to go this way though.